client_end_user/shared/src/commonMain/kotlin/presentation/auth/signup/registrationSubmit/RegistrationSubmitScreenModel.kt

```diff
@@ -0,0 +1,136 @@
+package presentation.auth.signup.registrationSubmit
+
+import cafe.adriel.voyager.core.model.coroutineScope
+import domain.usecase.IManageAuthenticationUseCase
+import domain.usecase.validation.IValidationUseCase
+import domain.utils.AuthorizationException
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import presentation.base.BaseScreenModel
+import presentation.base.ErrorState
+import presentation.base.ErrorState.InvalidEmail
+import presentation.base.ErrorState.InvalidFullName
+import presentation.base.ErrorState.InvalidPhone
+import presentation.base.ErrorState.NoInternet
+import presentation.base.ErrorState.RequestFailed
+import presentation.base.ErrorState.UserAlreadyExists
+import presentation.base.ErrorState.WifiDisabled
+
+class RegistrationSubmitScreenModel(
```

Comment:
as i remeber from uncle Bareq's review he asked us to try another way to pass the data between screens,
check "RestaurantInformationScreenModel" in resturant app to check this way and let's descuss

Reply:
https://github.com/TheChance101/beep-beep/blob/develop/client_restaurant/shared/src/commonMain/kotlin/presentation/information/RestaurantInformationScreen.kt

Same way I see

Reply:
i didn't get the meaning you means, but  look at the view model here :
https://github.com/TheChance101/beep-beep/blob/develop/client_restaurant/shared/src/commonMain/kotlin/presentation/information/RestaurantInformationScreenModel.kt
and eng Alhams use "  getRestaurantInfo(get())" to get the args and avoid to pass it in the constructor of view model like the comment of the review was

Reply:
and this in the screen itself :
" initScreen(getScreenModel<RestaurantInformationScreenModel> { parametersOf(id) })"

Reply:
I tried but it's not working with 2 params